packages/conform-zod/index.ts

```diff
@@ -96,36 +96,37 @@ export function getFieldsetConstraint<Source extends z.ZodTypeAny>(
 		'pattern',
 	];
 
-	function resolveFieldsetConstarint<T extends Record<string, any>>(
+	function resolveFieldsetConstraint<T extends Record<string, unknown>>(
 		schema: z.ZodType<T>,
 	): FieldsetConstraint<z.input<Source>> {
 		if (schema instanceof z.ZodObject) {
-			const result: FieldsetConstraint<z.input<Source>> = {};
+			const result: Record<string, FieldConstraint<unknown>> = {};
 
-			for (const [key, def] of Object.entries(schema.shape)) {
-				// @ts-expect-error
+			for (const [key, def] of Object.entries<z.ZodType<unknown>>(
+				schema.shape,
+			)) {
 				result[key] = inferConstraint(def);
 			}
 
-			return result;
+			return result as FieldsetConstraint<z.input<Source>>;
```

Comment:
and then assume we did our duediligence and hand back the type the caller is expecting.